### PR TITLE
Enable nvidia-persistenced and increase nvidia-smi timeout to 40s (from 25s)

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
To prevent timeout issues during CSE execution, for select GPU nodes

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by ganeshkumar5699** [comment]
Moved select changes from [henryli001](https://github.com/henryli001)'s PR: https://github.com/Azure/AgentBaker/pull/2829

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -564,7 +564,13 @@ configGPUDrivers() {
     if [[ $OS == $UBUNTU_OS_NAME ]]; then
         retrycmd_if_failure 120 5 25 nvidia-modprobe -u -c0 || exit $ERR_GPU_DRIVERS_START_FAIL
     fi
-    retrycmd_if_failure 120 5 25 nvidia-smi || exit $ERR_GPU_DRIVERS_START_FAIL
+    
+    # Noticed that NVIDIA driver version R515 and later versions will experience long loading time of nvidia-smi on 
+    # ND A100-v4 and NDm A100-v4 series. For now, use the workaround solution to enable NVIDIA persistence mode which 
+    # will accelerate the loading time of future nvidia-smi operations
+    retrycmd_if_failure 120 5 35 nvidia-persistenced --persistence-mode || exit $ERR_GPU_DRIVERS_START_FAIL
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_config.sh:571]: https://github.com/Azure/AgentBaker/pull/2829/files#r1143546871

not sure this is sufficient for user scenarios, it just makes provisioning work

File: e2e/cases.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ var cases = map[string]scenarioConfig{
 		bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
 			nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_NC6"
 			nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-18.04-gen2"
-			nbc.AgentPoolProfile.VMSize = "Standard_NC6"
+			nbc.AgentPoolProfile.VMSize = "Standard_ND96asr_v4"
```
[the remark this answers is not in the document]
Comment (ganeshkumar5699) [reply e2e/cases.go:17]: Yes, this is only for testing

**Comment by alexeldeib** [comment]
pulled into https://github.com/Azure/AgentBaker/pull/3209

**Pull request closed without merging**